encoded/encoded.go

```diff
@@ -38,4 +38,10 @@ type (
 		// Get extract the encoded values into strong typed value pointers.
 		Get(valuePtr ...interface{}) error
 	}
+
+	// Decoder is used to decode data
```

Comment:
I would not create a new interface as we already have Value and Values.
I would add
NewValue(data []byte) Value
NewValues(data[]) Values

Reply:
Good suggestion, changed.